dodo/search.py

```diff
@@ -42,17 +42,27 @@ class SearchModel(QAbstractItemModel):
     def __init__(self, q: str) -> None:
         super().__init__()
         self.q = q
+        self.d = []
+        self.json_str = ""
+        self.num_threads = 0
+        self.error_msg = None
         self.refresh()
 
     def refresh(self) -> None:
         """Refresh the model by (re-) running "notmuch search"."""
         logger.info("Beginning search refresh for '%s'", self.q)
         self.beginResetModel()
-        r = subprocess.run(['notmuch', 'search', '--format=json', self.q],
-                stdout=subprocess.PIPE)
-        self.json_str = r.stdout.decode('utf-8')
-        self.d = json.loads(self.json_str)
+        try:
+            r = subprocess.run(['notmuch', 'search', '--format=json', self.q],
+                    capture_output=True, check=True)
```

Comment:
While you're at it, with

```suggestion
                    capture_output=True, text=True, check=True)
```

you could avoid having to decode everything manually below.

Reply:
Done, thanks!